### ISSUE TITLE
[a11y] Tooltips throughout CiviForm are not accessible

### ISSUE THREAD

**Issue by melanie-exygy** [civiform/civiform#13444]
**Describe the bug**
The tooltips in CiviForm are not accessible to keyboard users, low mobility users and screen reader users.  It isn't possible to open the tooltip through keyboard navigation.  It only opens with a mouse click.  Also, the screen reader completely skips over the tooltip, so SR users won't even know that additional information exists.

Here is one of the tooltips (see the Administrative Identifier heading below).  There are others.

<img width="1037" height="960" alt="Image" src="https://github.com/user-attachments/assets/5cf5b79b-1dcb-4ad6-85a3-d45c67040f49" />

**WCAG Success Criterion**
[Criterion 2.1.1 - Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html)
[Criterion 1.1.1 - Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
and others

**To Reproduce**
1.  Log in as a CiviForm admin.
2. Click on the "Questions" tab.
3. Click on "Create new question".
4. Select any question type from the dropdown list.
5. Turn the screen reader on.
6. Attempt to use the Tab key to get to the "Info" icon next to "Administrative Identifier".
7. Note that you can't reach that icon and the text in the tooltip is never read.

**Expected behavior**
Keyboard-only and screen reader users should be able to access the information in the tooltip.

**Desktop (please complete the following information):**

- OS: MacOS
- Browser: Chrome
- Screen Reader (if applicable): VoiceOver

**Additional context**
We should start with an audit of tooltips in CiviForm.

#### Done when (design tasks):

What criteria need to be met for the issue to be finished?

- [x] Audit existing tooltips in the admin experiences
- [ ] Identify whether the tooltips can be converted to regular text to be more accessible (Ideally this is every instance; if not, record why a tooltip is needed and figure out accessibility for it)
- [ ] If needed, rewrite copy to better fit design solution. Copy should be written in plain language
- [ ] If needed, provided updated designs that include updated copy that has replaced tooltips

**Comment by myfriendgu** [comment]
@melanie-exygy are we able to follow how USWDS implements [tooltips](https://designsystem.digital.gov/components/tooltip/)? Here are their [accessibility test results](https://designsystem.digital.gov/components/tooltip/accessibility-tests/)

This essentially means that the info icon that serves as a tooltip needs to be treated as a button. LMK if that works, and I can design out different button states for it.

**Comment by melanie-exygy** [comment]
> [@melanie-exygy](https://github.com/melanie-exygy) are we able to follow how USWDS implements [tooltips](https://designsystem.digital.gov/components/tooltip/)? Here are their [accessibility test results](https://designsystem.digital.gov/components/tooltip/accessibility-tests/)
> 
> This essentially means that the info icon that serves as a tooltip needs to be treated as a button. LMK if that works, and I can design out different button states for it.

Yes, I think we could use the USWDS tooltip component.  It seems to solve the accessibility problems that I described above.  However, could we first consider not using a tooltip at all?  If the information in the tooltip is important, why would we hide it?  For example, in the admin question creation form, couldn't we add the tooltip info to the description text after "This value can't be changed later"?

**Comment by myfriendgu** [comment]
@melanie-exygy We can definitely do that; however, that would require a separate design ticket to do an audit* of the existing info tooltips for admin (I'm not sure if they exist on the applicant side?) and possibly another design ticket to rewrite the copy. FWIW, the copy one some of these information icons need to be reworked regardless of whether they're freestanding or as a tooltip, because they're pretty lengthy. I'm assuming that's the reason for having them behind a tooltip to begin with, but USWDS's guidelines is to keep tooltip copy short and sweet. I'll link those two tickets once I create them.

*Unless there's an easy way to do a tooltip scan on the engineering side so we know where all the tooltip are across the board?

**Comment by melanie-exygy** [comment]
> [@melanie-exygy](https://github.com/melanie-exygy) We can definitely do that; however, that would require a separate design ticket to do an audit* of the existing info tooltips for admin (I'm not sure if they exist on the applicant side?) and possibly another design ticket to rewrite the copy. FWIW, the copy one some of these information icons need to be reworked regardless of whether they're freestanding or as a tooltip, because they're pretty lengthy. I'm assuming that's the reason for having them behind a tooltip to begin with, but USWDS's guidelines is to keep tooltip copy short and sweet. I'll link those two tickets once I create them.
> 
> *Unless there's an easy way to do a tooltip scan on the engineering side so we know where all the tooltip are across the board?

@myfriendgu, unfortunately, I can't think of a way to do a scan for tooltips in the code since we don't have a dedicated tooltip component.  My intention in writing this ticket was to serve as a "home base" for the audit and re-design (or removal) of tooltips, so feel free to edit it and use it as a design ticket.  If you prefer to start fresh with two new tickets, that's fine too.

**Comment by myfriendgu** [comment]
@melanie-exygy Got it, I've updated the original ticket to add a done when section so it's easier to track in one ticket. Posting in this comment for anyone keeping up with the convo:

> #### Done when (design tasks):
> 
> What criteria need to be met for the issue to be finished?
> 
> - [ ] Audit existing tooltips in the admin experiences
> - [ ] Identify whether the tooltips can be converted to regular text to be more accessible (Ideally this is every instance; if not, record why a tooltip is needed and figure out accessibility for it)
> - [ ] If needed, rewrite copy to better fit design solution. Copy should be written in plain language

**Comment by myfriendgu** [comment]
To that end, I've started the audit. Spreadsheet can be accessed here: https://docs.google.com/spreadsheets/d/16BlFX2yYrtQ6eBmK6KKSq6cJbHeXWnAGufw28lLMwV0/edit?gid=0#gid=0

**Comment by myfriendgu** [comment]
Commenting here, but i've tagged @melanie-exygy in the spreadsheet above:

My suggestion for all of these is to simplify the tooltip content and convert them into the help text field under the form field header. the problem is, I don't believe that hep text field exists in our current build. Moving it to USWDS, I've included the help text so each form field has a heading and optional help text. not sure how we want to integrate that with the existing USWDS build?

<img width="1207" height="210" alt="Image" src="https://github.com/user-attachments/assets/6dac82e6-b1c0-4e8a-88a2-e6e850f75c9b" />

**Comment by melanie-exygy** [comment]
> Commenting here, but i've tagged [@melanie-exygy](https://github.com/melanie-exygy) in the spreadsheet above:
> 
> My suggestion for all of these is to simplify the tooltip content and convert them into the help text field under the form field header. the problem is, I don't believe that hep text field exists in our current build. Moving it to USWDS, I've included the help text so each form field has a heading and optional help text. not sure how we want to integrate that with the existing USWDS build?
> 
> <img alt="Image" width="1207" height="210" src="https://private-user-images.githubusercontent.com/33872753/627939274-6dac82e6-b1c0-4e8a-88a2-e6e850f75c9b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODUyNjI2NTgsIm5iZiI6MTc4NTI2MjM1OCwicGF0aCI6Ii8zMzg3Mjc1My82Mjc5MzkyNzQtNmRhYzgyZTYtYjFjMC00ZThhLTg4YTItZTZlODUwZjc1YzliLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA3MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNzI4VDE4MTIzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFjOTU0Zjg2NjNmNTk0YTJkOTJmN2I0YmM1NDAxNGM1MTViZGI1YWU3ZmQwZmZlYmVmOTFjY2E0MzIxZjA5MTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.VI_sxPP5vWbYSIwATBiY58BLWXcxN4TZfae7JoNbXdg">

@myfriendgu for the two input fields, we can always add a description (what you're calling "help text") and use `aria-describedBy`, whether or not it's a USWDS input field.  

For the table headers on the Reporting page, if we feel like the headers themselves aren't clear enough, I suggest that we add a sort of glossary either above or below the table explaining what each header means.  Tables are already tricky to make accessible, so that's why I don't recommend adding the info text to the table column headers.
